source/designers-guide/smarty-plugins/index.md

```diff
@@ -0,0 +1,257 @@
+
+Without the plugin, the call looks like this:
+
+Example: *normal style sheet call*
+```
+<link rel="stylesheet" media="screen, projection" href="http://www.meinshop.de/shopware/templates/my_template/frontend/_resources/styles/my_styles.css" />
+```
+
+
+By using the link plugin you only have to define the path relative to your own theme directory.
+
+Example: *Calling a style sheet via link plugin*
+```
+<link rel="stylesheet" media="screen, projection" href="{link file='frontend/_resources/styles/my_styles.css'}" />
+```
+
+The syntax of the plugin looks like this:
+
+``` 
+{* Syntax *}
+{link file="[PATH TO FILE]"}
+```
+
+## Media plugin ##
+
+With Shopware 5.2 the new media plugin was added.
+
+In your Smarty templates you may use the {media path=...} expression to get the fully qualified URL.
+
+Example: *Calling an image via media plugin* 
+```
+<img src="{media path="media/image/my-fancy-image.png"}">
+``` 
+ 
+{media} evaluates the given path at template's compile time, so you cannot use runtime variables for its path argument (generally you will use a constant path as in the example above). 
+ 
+[Shopware 5 Media Service Documentation](https://developers.shopware.com/developers-guide/shopware-5-media-service/#url-generation)
+
+## Url plugin ##
+
+The url plugin gathers the URLS together from throughout the frontend and passes those specified on to the appropriate controller and action. Let's assume that you have a link that should connect to the wish list, then the call would look like this:
+
+ 
+Example: *example of a call to the wish list*
+``` 
+{* A link to notes *}
+<a href="{url controller='note'}" title="Show notes">
+	Show notes
+</a>
+``` 
+ 
+The index action is always the default action. If you wish to call, for example, instant downloads in the account area, then the action must be passed on accordingly.
+
+Example: *example of a call to instant downloads*
+``` 
+{* A link to instant downloads *}
+<a href="{url controller='account' action='downloads'}" title="Open instant downloads">
+	Open instant downloads
+</a>
+``` 
+ 
+You also have the option to pass on to the controller parameters. For this example, we'll extend the call to instant downloads with the parameter sParam with the value test.
+
+Exmaple: *call to instant download with parameter*
+``` 
+{* A link to instant downloads *}
+<a href="{url controller='account' action='downloads' sParam='test'}" title="Open instant downloads with parameter">
+	Open instant downloads with parameter
+</a>
+``` 
+ 
+The syntax of the url plugin is as follows:
+ 
+ 
+Example: *Syntax url plugin*
+``` 
+{* Syntax *}
+{url controller='[CONTROLLERNAME]' action='[ACTIONNAME]' [MORE PARAMETERS='PARAMETERVALUE']}
```

Comment:
the `module` parameter is missing here. it defaults to `frontend` but `widgets` is also a use case